changelog/fragments/k8s-1.24-bump.yaml

```diff
@@ -0,0 +1,110 @@
+# entries is a list of entries to include in
+# release notes and/or the migration guide
+
+# TODO(everettraven): update changelog
+entries:
+  - description: >
+      For Go (go/v3) and Helm Hybrid (hybrid) Operators: Upgrade Kubernetes 1.23 to 1.24
+
+    # kind is one of:
+    # - addition
+    # - change
+    # - deprecation
+    # - removal
+    # - bugfix
+    kind: "change"
+
+    # Is this a breaking change?
+    breaking: false
+
+    # Migration can be defined to automatically add a section to
+    # the migration guide. This is required for breaking changes.
+    migration:
+      header: Go (go/v3) and Hybrid Helm (hybrid) Operators - Upgrading to Kubernetes 1.24
+      body: |
+        To update your project to support Kubernetes 1.24:
+        
+        1. In `go.mod`:
+
+        - Replace `k8s.io/api v0.23.5` with `k8s.io/api v0.24.0`
+        - Replace `k8s.io/apimachinery v0.23.5` with  `k8s.io/apimachinery v0.24.0`
+        - Replace `k8s.io/client-go v0.23.5` with `k8s.io/client-go v0.24.0`
+        - Replace `sigs.k8s.io/controller-runtime v0.11.2` with `sigs.k8s.io/controller-runtime v0.12.1`
+        - Hybrid Helm (hybrid) only: Replace `github.com/operator-framework/helm-operator-plugins v0.0.11` with `github.com/operator-framework/helm-operator-plugins v0.0.12-0.20220608155702-d3967d2ae2ac`
+
+        2. Run `go mod tidy` to ensure all dependencies are upgraded
+        
+        3. In `Makefile`:
+        - Replace `CONTROLLER_TOOLS_VERSION ?= 0.8.0` with `CONTROLLER_TOOLS_VERSION ?= 0.9.0`
```

Comment:
This one is no related to k8s dep

Reply:
Controller tools doesn't use k8s 1.24 dependencies until v0.9.0 which makes me feel as though it should be included in the migration guide to supporting k8s 1.24.

https://github.com/kubernetes-sigs/controller-tools/releases/tag/v0.9.0